src/components/NewDiskDialog/index.js

```diff
@@ -0,0 +1,315 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { Modal } from 'patternfly-react'
+
+import { createDiskForVm, cleanNewDiskDialogSubtree } from './actions'
+import { msg } from '../../intl'
+import { logError } from '../../helpers'
+import SelectBox from '../SelectBox'
+import { flatMap } from '../../utils'
+import style from './style.css'
+
+class NewDiskDialog extends React.Component {
+
+  constructor (props) {
+    super(props)
+    props.cleanStoreSubtreeFunction()
+    this.state = {
+      storageDomainId: undefined,
+      alias: undefined,
+      sizeGb: '1',
+    }
+
+    this.onStorageDomainChange = this.onStorageDomainChange.bind(this)
+    this.onCreate = this.onCreate.bind(this)
+    this.onAliasChange = this.onAliasChange.bind(this)
+    this.onSizeChange = this.onSizeChange.bind(this)
+    this.onCancel = this.onCancel.bind(this)
+  }
+
+  componentWillMount () {
+    this.setState({
+      storageDomainId: this.getPreselectedStorageDomainId(),
+      alias: this.getNextAliasSuggestion(),
+    })
+  }
+
+  componentWillReceiveProps (newProps) {
+    if (newProps.done) {
+      this.props.onClose()
+    }
+  }
+
+  getDataCenterId () {
+    const clusterId = this.props.vm.cluster.id
+    const cluster = Object.values(this.props.clusters.clusters)
+      .filter(cluster => cluster.id === clusterId)[0]
+    const dataCenterId = cluster && cluster.dataCenterId
+    if (!dataCenterId) {
+      logError('No DataCenter found for VM=', this.props.vm, 'provided clusters were', this.props.clusters)
+      throw new Error()
```

Comment:
Please add comment here, that this can be caused by corrupted data only.
please comment, the `throw new Error()` is for logging stack-trace *only*

Reply:
reworked